src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -411,11 +321,11 @@ public virtual void AuthenticateAsClient(string targetHost, X509CertificateColle
         public void AuthenticateAsClient(SslClientAuthenticationOptions sslClientAuthenticationOptions)
         {
             ArgumentNullException.ThrowIfNull(sslClientAuthenticationOptions);
+            ArgumentNullException.ThrowIfNull(sslClientAuthenticationOptions.TargetHost, nameof(sslClientAuthenticationOptions.TargetHost));
```

Comment:
The name in the exception here will be just "TargetHost".  Is that the intent?

Reply:
This is what the old code did. I think it is ok here since it is name of property that cannot be null. If you have suggestion how to improve wording, please let me know.

Reply:
Just to be clear, I mean the text will literally be "TargetHost"; it won't be "sslClientAuthenticationOptions.TargetHost".

Reply:
It would be functional change (I did not originally intended) but we can also not throw and treat null and empty string equally. If we want to update the message would "SslClientAuthenticationOptions.TargetHost" be sufficient or should I add better message to the resource file?